### ISSUE TITLE
tests: change the color of skipped functional tests

### ISSUE THREAD

**Issue by 0xB10C** [bitcoin/bitcoin#24791]
The dark gray color used to mark skipped functional tests is somewhat unfortunate. I don't see skipped tests at all with my terminal settings unless I mark the text. For others, dark gray/black is hard to read on a gray background. 

My terminal:

![image](https://user-images.githubusercontent.com/19157360/162059456-7a21da0d-8a6c-4d03-ad2a-261e0c7209ec.png)

![image](https://user-images.githubusercontent.com/19157360/162059921-9d5a1b19-4198-4b82-aa22-bb50c03a0c0d.png)

And @jacobpfickes terminal from [twitter](https://twitter.com/cryptocowboy__/status/1511505046378266627) which reminded me to open this good first issue.
![image](https://user-images.githubusercontent.com/19157360/162060107-e3188d00-4226-4822-ac6e-29e0ec225302.png)

For example, during review it's important to see which tests were skipped. It could happen that you think you've run the tests for something, but you in fact haven't and the tests were skipped, but you just don't notice.

A solution would be to use a yellow/orange color to mark skipped tests. Green for pass, red for fail and yellow/orange for skipped. Another easy solution would be for me to change my terminal settings, however I think the gray text on gray background could affect more devs than just me.

For reference, the colored functional tests output was introduced in https://github.com/bitcoin/bitcoin/pull/10159.

#### Useful skills:

Python and running the functional tests. 

#### Want to work on this issue?

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

**Comment by laanwj** [comment]
FWIW, a common way to avoid these kind of problems is to make sure all your 8/16 terminal colors have a good contrast against your background.  It's a limited palette so having any unusable colors is a waste. The default color scheme tends to be most suitable for white backgrounds (usually).

That said, I'm not against changing the colors but it seems like a literal shed-painting issue, there's always going to be someone with a background shade / palette combination that will have trouble seeing some color.

**Comment by jacobpfickes** [comment]
That is a good point about never being able to satisfy every user's needs. I went ahead and made the change to yellow if the team decides it's a better option (I don't think it looks too bad on a white background but I understand it's not optimal). On the other hand, maybe another color that works well on both white and black backgrounds would satisfy the majority of users and could be a better option.

**Comment by MarcoFalke** [comment]
I am using the default settings on my OS and I am also seeing the skipped tests very mute. No opinion on whether that needs fixing or how to fix it.

**Comment by 0xB10C** [comment]
Thanks for the input! I agree that we don't want a PR to change the color every other month. However, I do think the status quo can be improved upon.

Another solution could be to remove the coloring of the skipped tests entirely and print them as normal, uncolored text. This should bring the best readability and avoid any future shed re-painting.

**Comment by MarcoFalke** [comment]
> Another solution could be to remove the coloring of the skipped tests entirely and print them as normal, uncolored text. This should bring the best readability and avoid any future shed re-painting.

Yeah, just using the default color, but non-bold should work just fine.

**Comment by laanwj** [comment]
Speaking of bold, we also shouldn't be making the assumption in scripts that bold means bright. This is something the Linux terminal historically does, but many other terminals don't (or have configurable). So black is really black in that case.

**Comment by TokensDL** [comment]
> Màu xám đậm được sử dụng để đánh dấu các bài kiểm tra chức năng bị bỏ qua là hơi đáng tiếc. Tôi hoàn toàn không thấy các bài kiểm tra bị bỏ qua với cài đặt đầu cuối của mình trừ khi tôi đánh dấu văn bản. Đối với những người khác, màu xám đậm / đen khó đọc trên nền xám.
> 
> Thiết bị đầu cuối của tôi:
> 
> ![hình ảnh](https://user-images.githubusercontent.com/19157360/162059456-7a21da0d-8a6c-4d03-ad2a-261e0c7209ec.png)
> 
> ![hình ảnh](https://user-images.githubusercontent.com/19157360/162059921-9d5a1b19-4198-4b82-aa22-bb50c03a0c0d.png)
> 
> Và thiết bị đầu cuối @jacobpfickes từ [twitter](https://twitter.com/cryptocowboy__/status/1511505046378266627) đã nhắc tôi mở số đầu tiên hay này. ![hình ảnh](https://user-images.githubusercontent.com/19157360/162060107-e3188d00-4226-4822-ac6e-29e0ec225302.png)
> 
> Ví dụ, trong quá trình xem xét, điều quan trọng là phải xem những bài kiểm tra nào đã bị bỏ qua. Có thể xảy ra trường hợp bạn nghĩ rằng bạn đã chạy các bài kiểm tra cho một thứ gì đó, nhưng thực tế là bạn đã không chạy và các bài kiểm tra đã bị bỏ qua, nhưng bạn không nhận thấy.
> 
> Một giải pháp sẽ là sử dụng màu vàng / cam để đánh dấu các bài kiểm tra bị bỏ qua. Màu xanh lá cây cho điểm đạt, màu đỏ cho điểm trượt và màu vàng / da cam cho trường hợp bỏ qua. Một giải pháp dễ dàng khác dành cho tôi là thay đổi cài đặt thiết bị đầu cuối của mình, tuy nhiên tôi nghĩ văn bản màu xám trên nền xám có thể ảnh hưởng đến nhiều nhà phát triển hơn chỉ tôi.
> 
> Để tham khảo, đầu ra các bài kiểm tra chức năng có màu đã được giới thiệu trong [# 10159](https://github.com/bitcoin/bitcoin/pull/10159) .
> 
> #### Kỹ năng hữu ích:
> Python và chạy các bài kiểm tra chức năng.
> 
> #### Bạn muốn làm việc về vấn đề này?
> Để được hướng dẫn về cách đóng góp, vui lòng đọc [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) trước khi mở yêu cầu kéo của bạn.

**Comment by CryptoUser243** [comment]
TL;DR: This issue degrades the UX of the report for a minority of users. The suggest resolution #24793 slightly degrades the UX of the remaining users. Perhaps its worth exploring whether @laanwj's suggestion of ```\033[0;90m``` would resolve the issue whilst minimising the impact on unaffected users before moving ahead with #24793?

Given that the potential for 'shed re-painting' has already been raised by other users, perhaps a more detailed exposition of the factors at play, written as a set of challengable assertions and assumptions, would aid any further discussion. See below if you are interested:

### On the purpose of the report and how the colour of the skipped items help to achieve this

The objectives of the report in order of priority:
1. state whether the test run passed or failed
1. list failed test - to show the user which tests should be explored further to understand what went wrong
1. list passed test - so the user can confirm that the expected tests actually ran and passed
1. highlight how long each test, and the test run in total, took so that they can identify when tests are running unusually slow
1. list skipped test - so the user can see if they neglected to run a test they intended to

Three UI tools are used in the in are used to indicate the result:
1. colour (currently Green/Red/Grey)
1. text label (Passed/Failed/Skipped)
1. icon

The text label provides the clearest indication of the outcome of the test but is only ever read peripherally as this information is embedded directly onto the name of test through its colour.

The pass/fail to green/red colour mapping are a universal standard in CLI test reports and are therefore intuitive to most users.

The skipped to 'muted standard text colour' colour mapping is widely, but not universal, used in CLI test reports. 'Standard body text colour' and yellow are other colours sometimes used.

### On the impact of the issue

For a minority of users it appears as illegibly close to the background colour but different enough to make it clear that there is text present.

Workaround - most user can make it legible simply by selecting the text. This is likely to be intuitive to most to CLI users.

This provides a poor UX 1) for a nice to have feature of the report and 2) the general usage of the report as the interface is jarring and strains the eyes.

### On the cause of the issue:

Terminals vary in their default background colour and their interpretation of colour values. Most allow these setting to be customised.
When CLIs use colour there is no guarantee that their display will meet access ability standards.

### In defense of the status quo

For the majority of users the colour chosen for skipped items appears as 'muted standard text colour'. This is most appropriate colour for 2 reasons:
1. it is commonly used standards for CLI test run reports
1. the skipped tests are the least important information to communicate so de-emphasising them improves the usability of the report

The use of ```\033[1;30m'``` as a muted colour should work across all sensibly configured terminals. If this isn't the case then a bug should be raised with the terminal vendor.

Some users (eg. @ayush933 and @kristapsk) have suggested #24793 (using the default colour) degrades the usability of the report.

This issue only affects a minority of users assumed to be 2-5%.

### Arguments for using the default text color as proposed in #24793:

Due to the variety of colour interpretation in CLIs colours should be used sparingly. The slight usability benefits of using grey are outweighed by the significant degradation for the minority of users.

### Arguments for searching for an alternative

@laanwj suggestion of ```\033[0;90m``` may resolve the issue without degrading the reports usability for other users.

### Recommendation

Explore whether ```\033[0;90m``` is preferable to #24793 for both affected and unaffected users. If it is not: fallback to #24793 as the severity of the usability impact on affected users (assumed to be 2-5% of the user-base) outweighs the minor usability degradation for unaffected users.

**Comment by TokensDL** [comment]
La gi

**Comment by laanwj** [comment]
Thanks @CryptoUser243. I went with merging #24793 as IMO "unreadable" complaints weigh stronger than "it stands out too much". It gets rid of the `GREY = "\033[1;30m"` (which is bold black, not grey), so is more terminal-neutral.
